pkg/detector/policy.go

```diff
@@ -175,6 +176,90 @@ func (d *ResourceDetector) getAndApplyClusterPolicy(object *unstructured.Unstruc
 	return d.ApplyClusterPolicy(object, objectKey, matchedClusterPropagationPolicy)
 }
 
+// removeSpecificResourceAndRBLabels remove the propagation policy labels of specific resource and corresponding binding.
+func (d *ResourceDetector) removeSpecificResourceAndRBLabels(resource *unstructured.Unstructured, removeLabels []string) error {
+	// 1. get rb of given resource
+	rbName := names.GenerateBindingName(resource.GetKind(), resource.GetName())
+	rb := &workv1alpha2.ResourceBinding{}
+	if err := d.Client.Get(context.TODO(), client.ObjectKey{Namespace: resource.GetNamespace(), Name: rbName}, rb); err != nil {
+		klog.Errorf("Failed to get ResourceBinding: %s, err: %+v", rbName, err)
+		return err
+	}
+
+	// 2. check whether rb related resource match to the given resource.
+	// the mismatch would never happen under normal circumstances, if it really happened, just print logs and return nil.
+	// this logic is also intended to be compatible with previous versions which talked about the issue of Ingress(networking.k8s.io/v1) and Ingress(extensions/v1beta1).
+	if rb.Spec.Resource.APIVersion != resource.GetAPIVersion() || rb.Spec.Resource.Kind != resource.GetKind() ||
+		rb.Spec.Resource.Name != resource.GetName() || rb.Spec.Resource.Namespace != resource.GetNamespace() {
+		klog.Errorf("The rb (%s) related resource is not match to the given resource: %s/%s", rbName, resource.GetNamespace(), resource.GetName())
+		return nil
+	}
+
+	// 3. clean the labels in binding
+	bindingCopy := rb.DeepCopy()
+	for _, label := range removeLabels {
```

Comment:
```suggestion
	util.RemoveLabels(bindingCopy, removeLabels...)
```

Reply:
Are you sure ???

pay attention to the function declaration `func RemoveLabels(obj *unstructured.Unstructured, labelKeys ...string)`

Reply:
The current implementation only has two lines. It seems that there is no need to extract a function. You can ignore this suggestion.